packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -283,13 +297,34 @@ function startMeasuringDocumentElement(): ThunkAction<() => void, unknown, unkno
   }
 }
 
+function startHandlingKeyDownEvent(): ThunkAction<() => void, State, unknown, Action> {
+  return (_dispatch, getState) => {
+    window.document.body.addEventListener('keydown', handle)
+
+    return () => {
+      window.document.body.removeEventListener('keydown', handle)
+    }
+
+    function handle(event: KeyboardEvent) {
+      if (ReactPage.getBuilderEditMode(getState()) !== BuilderEditMode.BuilderEditMode.INTERACT) {
+        return
+      }
+
+      if (isHotkey('escape')(event)) {
```

Comment:
We might also want to check if the default behavior has been prevented, and if so, not go back to build mode.

Something else to consider, instead of focusing the parent, do we want to send a message to the builder so that it switches to build mode immediately?